js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,194 @@
+/* TASK 1
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+*
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+*                            Если длина 10 -> 'you win'
+* */
+function showMeType(arg) {
+	let res;
+	//res = typeof arg === 'number' || typeof arg === 'object' ? true : false;
+	//return res;
+	if (typeof arg === 'number' || typeof arg === 'object') {
+		res = true;
+	};
+
+	if (typeof arg === 'function') {
+		res = false;
+	};
+
+	if (typeof arg === 'string' && arg.length != 10) {
+		res = `Длина Вашей строки: ${arg.length}`;
+	} else if (arg.length == 10) {
+		res = `you win`;
+	}
+
+	return res;
+}
+
+function helper(){
+	console.log('I am a helper');
+};
+
+console.log('task 1-1 ---->', showMeType(6));//true
+console.log('task 1-2 ---->', showMeType([]));//true
+console.log('task 1-3 ---->', showMeType({}));//true
+console.log('task 1-4 ---->', showMeType(helper));//false
+console.log('task 1-5 ---->', showMeType('012345'));//6
+console.log('task 1-6 ---->', showMeType('0123456789'));//you win
+
+//--------------------------------------------------------------------------------------------
+
+/* TASK 2
+ 1. Напишите функцию которая принимает 2 числа
+ и возвращает массив содержащий числа между первым числом и вторым числом;
+ */
+
+function numbersBetween(a, b) {
+	let arr = [];
+	for (let i = a; i <= b; i++) {
+		arr[arr.length] = i;
+	}
+	return arr;
+}
+
+console.log('task 2-1 ---->', numbersBetween(3, 5));// 3, 4, 5
+console.log('task 2-2 ---->', numbersBetween(10, 12));// 10, 11, 12
+console.log('task 2-3 ---->', numbersBetween(4, 15));// 4,5,6,7,8,9,10,11,12,13,14,15
+
+//------------------------------------------------------------------------------------------------
+
+/* TASK 3
+ 2. Перепишите задачу FizzBuzz, но с использованием цикла.
+ Расчет чисел должен идти до 100
+ */
+
+// 1. FizzBuzz 3, 5, 3 && % 5
+
+function FizzBuzz(num) {
+	let res = '';
+	/*if (num % 3 == 0) {
+		res += 'Fizz';
+	};
+	if (num % 5 == 0) {
+		res += 'Buzz';
+	};
+	if (res.length == 0){
+		return num;
+	}
+	return res;*/
+	res += (num % 3 == 0) ? 'Fizz' : '';
```

Comment:
wow :)
first 2 if statements were fine :)

I've thought you would update only last return

Reply:
But is it ok?

Reply:
not really, it looks weird. it looks like 
"oh wow never seen such syntax before" :)